checkers: boolExprSimplify false positive for floats

Triggered at https://github.com/flyingmutant/rapid/blob/master/utils.go#L29.

```
./src/github.com/flyingmutant/rapid/utils.go:29:9: boolExprSimplify: can simplify `p > 0 && p <= 1` to `p == 1`
```
Suggested replacement only makes sense for int values.
Referenced code uses floating point values.

--- a/checkers/boolExprSimplify_checker.go
+++ b/checkers/boolExprSimplify_checker.go
@@ -235,6 +235,10 @@ func (c *boolExprSimplifyChecker) removeIncDec(cur *astutil.Cursor) bool {
 }
 
 func (c *boolExprSimplifyChecker) foldRanges(cur *astutil.Cursor) bool {
+	if c.hasFloats { // See #673
+		return false
+	}
+
 	e, ok := cur.Node().(*ast.BinaryExpr)
 	if !ok {
 		return false

--- a/checkers/boolExprSimplify_checker.go
+++ b/checkers/boolExprSimplify_checker.go
@@ -235,7 +235,7 @@ func (c *boolExprSimplifyChecker) removeIncDec(cur *astutil.Cursor) bool {
 }
 
 func (c *boolExprSimplifyChecker) foldRanges(cur *astutil.Cursor) bool {
-	if c.hasFloats { // See #673
+	if c.hasFloats { // See #848
 		return false
 	}
 

--- a/checkers/testdata/boolExprSimplify/negative_tests.go
+++ b/checkers/testdata/boolExprSimplify/negative_tests.go
@@ -58,6 +58,9 @@ func floatCompare() {
 
 	// Can't be simplified to `f1 != f2`.
 	_ = !(f1 == f2)
+
+	// Can't be simplified to `p == 1`.
+	_ = f1 > 0 && f1 <= 1
 }
 
 func balancedIncDec(x, y, z int) {

